cedar-lean/Cedar/Validation/RequestEntityValidator.lean

```diff
@@ -0,0 +1,157 @@
+import Cedar.Validation.Validator
+import Cedar.Validation.Typechecker
+namespace Cedar.Validation
+
+open Cedar.Spec
+open Cedar.Data
+
```

Comment:
What's the relation of the definitions in this file to the `RequestAndEntitiesMatchEnvironment` predicate used in proofs?

Reply:
it's almost a reflection from prop to bool, except that the function for entity validation has a pre-processing step that adds entries in the action schema to the list of entities and the entity schema, to match both the rust code and the preconditions for proofs.

Reply:
As discussed in person: it would be awesome if we could reuse the `Prop` definitions here instead of copying them with the `Bool` type

Reply:
will make a new PR for proofs of this